src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithPlainTextCommentFilter.java

```diff
@@ -0,0 +1,339 @@
+
+/**
+ * <p>
+ *     A filter that uses comments to suppress audit events.
+ *     The filter can be used only to suppress audit events received from
+ *     {@link com.puppycrawl.tools.checkstyle.api.FileSetCheck} checks.
+ *     SuppressWithPlainTextCommentFilter knows nothing about AST,
+ *     it treats only plain text comments and extracts the information required for suppression from
+ *     the plain text comments. Currently the filter supports only single line comments.
+ * </p>
+ * <p>
+ *     Rationale:
+ *     Sometimes there are legitimate reasons for violating a check. When
+ *     this is a matter of the code in question and not personal
+ *     preference, the best place to override the policy is in the code
+ *     itself.  Semi-structured comments can be associated with the check.
+ *     This is sometimes superior to a separate suppressions file, which
+ *     must be kept up-to-date as the source file is edited.
+ * </p>
+ * @author Andrei Selkin
+ */
+public class SuppressWithPlainTextCommentFilter extends AutomaticBean implements Filter {
+
+    /** Comment format which turns checkstyle reporting off. */
+    private static final String DEFAULT_OFF_FORMAT = "// CHECKSTYLE:OFF";
+
+    /** Comment format which turns checkstyle reporting on. */
+    private static final String DEFAULT_ON_FORMAT = "// CHECKSTYLE:ON";
+
+    /** Default check format to suppress. By default the filter suppress all checks. */
+    private static final String DEFAULT_CHECK_FORMAT = ".*";
+
+    /** Regexp which turns checkstyle reporting off. */
+    private Pattern offCommentFormat = CommonUtils.createPattern(DEFAULT_OFF_FORMAT);
+
+    /** Regexp which turns checkstyle reporting on. */
+    private Pattern onCommentFormat = CommonUtils.createPattern(DEFAULT_ON_FORMAT);
+
+    /** The check format to suppress. */
+    private String checkFormat = DEFAULT_CHECK_FORMAT;
+
+    /** The message format to suppress.*/
+    private String messageFormat;
+
+    /**
+     * Sets an off comment format pattern.
+     * @param pattern off comment format pattern.
+     */
+    public final void setOffCommentFormat(Pattern pattern) {
+        offCommentFormat = pattern;
+    }
+
+    /**
+     * Sets an on comment format pattern.
+     * @param pattern  on comment format pattern.
+     */
+    public final void setOnCommentFormat(Pattern pattern) {
+        onCommentFormat = pattern;
+    }
+
+    /**
+     * Sets a pattern for check format.
+     * @param format pattern for check format.
+     */
+    public final void setCheckFormat(String format) {
+        checkFormat = format;
+    }
+
+    /**
+     * Sets a pattern for message format.
+     * @param format pattern for message format.
+     */
+    public final void setMessageFormat(String format) {
+        messageFormat = format;
+    }
+
+    @Override
+    public boolean accept(AuditEvent event) {
+        boolean accepted = true;
+
+        if (event.getLocalizedMessage() != null) {
+
```

Comment:
remove excess space.
Keep an eye for changes coming in https://github.com/checkstyle/checkstyle/pull/5312 .

Reply:
Done